### DISCUSSION TITLE
v1.0.2

### DISCUSSION THREAD

**Discussion by JacobsenKim** [JacobsenKim/ha2mariadb#6] [Announcements]
2024-04-17 - Update to work with the new schema version 43.

**Full Changelog**: https://github.com/JacobsenKim/ha2mariadb/compare/v1.0.1...v1.0.2

**Discussions**: https://github.com/JacobsenKim/ha2mariadb/discussions/3

<hr /><em>This discussion was created from the release <a href='https://github.com/JacobsenKim/ha2mariadb/releases/tag/v1.0.2'>v1.0.2</a>.</em>

**Comment by NicolasGoeddel** [comment]
In your how-to you never explain what scripts should be used at which time. Also they will not work in my setup since I run HA and MariaDB in a docker compose stack.
But I am interested in the stuff you wrote for creating missing indices. Does your scripts also set the auto increment values properly? And why do you split stuff at 50000 lines? My Dump consists of 31,323,815 lines and this would mean it splits it a lot.

**Reply by JacobsenKim** [reply]
all is in the wiki step by step, copy paste to the console.
https://github.com/JacobsenKim/ha2mariadb/wiki/From-Ubuntu

I dont see why it wont work in your docker setup, the script connect to mariaDB and write to the DB so if you insert the correct login info then it should work.


i dont understand this part

"But I am interested in the stuff you wrote for creating missing indices"

Can you link to the part of code ?

auto increment values you have to enter manuel as described in the wiki, but the script offer to show you the numbers you have to type in the z9_grep_mysql_importme_last_43.sql file before you do the last import.

I split the data files so its possible to look inside with a txt app or excel, i add it when i was working on the method, its not necessary but good to have in case something goes wrong or mariaDB dont like your big file and need to spilt it, you can change the numbers in the script
As described in wiki 
"
This will create an output folder with all files, the split folder is the big files over 50000 lines split in to small files if neded.

The big files is still in the output folder and is the one that is imported.
"

**Reply by NicolasGoeddel** [reply]
Hi,
sorry, I was just dumb. I didn't know how to use the wiki properly. 🙈
I now found all the information in it and I am already working on a more generic script. I can send the result to you if you want to have a look at it.

**Reply by JacobsenKim** [reply]
Nice, yes ill have a look at it.

**Reply by NicolasGoeddel** [reply]
I am bit confused about the schema part when I compare the SQLite output of my installation with the MySQL-Schema you provide in your repository.
For example:
- events.event_type is `VARCHAR(64)` in SQLite, but `char(0)` in MySQL?
- recorder_runs.start is `DATETIME` in SQLite, but `datetime(6) NOT NULL` in MySQL and at the same time
- recorder_runs.end is also `DATETIME` in SQLite, but `datetime(6) DEFAULT NULL` in MySQL.
- statistics.created is `DATETIME` in SQLite, but `char(0)` in MySQL.

From where do you have extracted the MySQL schema? It's confusing that dates are now written to strings with zero length and stuff like that. :thinking:

**Reply by NicolasGoeddel** [reply]
I think I found out why some columns have the datatype `char(0)`. There are just legacy columns. Here's just one example for that in the original source: https://github.com/home-assistant/core/blob/10a2fd7cb66c2d39c2652c9b85f78e89a0dfc7a9/homeassistant/components/recorder/db_schema.py#L425

**Reply by JacobsenKim** [reply]
I have no clue why the 2 DB use different naming and size, just make sure the order in the table match your data set.


I got the sql file from mariaDB in a new install homeassistant and made the dump with phpmyadmin, then delete the data and split the sql file in 2, a first for the tables and the second for the key and stuff after the data is add to the DB.

from the sql file


"
-- phpMyAdmin SQL Dump
-- version 5.2.1
-- https://www.phpmyadmin.net/
--
-- Host: core-mariadb:3306
-- Generation Time: Apr 17, 2024 at 03:41 PM
-- Server version: 10.11.6-MariaDB
-- PHP Version: 8.1.27
"

**Reply by NicolasGoeddel** [reply]
I see. Well, I looked in the source code and the SQL statements you've provided in your repository. My schema conversion script works fine now. I was able to do everything with Bash. I also was able to make the import of big databases a bit faster. Mine already has around 30 million rows which takes a while.
I soon should be ready to show you the new migration script. I just want to make it a bit more convenient to use now.

**Reply by NicolasGoeddel** [reply]
Yesterday I migrated my SQLite database to MariaDB using my script. I took a while but went fully automatically.
At the moment I am in the process of refactoring it a bit, but if you want you can have a first look at it. I just created a repository for it and filled it with life: https://github.com/NicolasGoeddel/ha-sqlite2mariadb

The README.md does not have many explanations yet and the whole thing is only able to do it fully automatic if you run HA and the database using `docker compose` but I started adding the possibility to do it with other installation methods too.
Just have a look into `migrate.sh` and `migration.conf.sh` if you are curious.
